### ISSUE TITLE
[ceskatelevize] youtube_dl.utils.ExtractorError: No video formats found

### ISSUE THREAD

**Issue by mcepl** [ytdl-org/youtube-dl#29415]
## Checklist
- [x] I'm reporting a broken site support
- [x] I've verified that I'm running youtube-dl version **2021.06.06**
- [x] I've checked that all provided URLs are alive and playable in a browser
- [x] I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [x] I've searched the bugtracker for similar issues including closed ones

## Verbose log

```
usansky$ youtube-dl --verbose https://www.ceskatelevize.cz/porady/10597292024-filmovy-dobrodruh-karel-zeman/31429434015/
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['--verbose', 'https://www.ceskatelevize.cz/porady/10597292024-filmovy-dobrodruh-karel-zeman/31429434015/']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, pref UTF-8
[debug] youtube-dl version 2021.06.06
[debug] Python version 3.8.10 (CPython) - Linux-5.12.12-1-default-x86_64-with-glibc2.2.5
[debug] exe versions: ffmpeg 4.4, ffprobe 4.4
[debug] Proxy map: {}
[CeskaTelevizePorady] 31429434015: Downloading webpage
[CeskaTelevize] iFramePlayer.php: Downloading webpage
[CeskaTelevize] iFramePlayer.php: Downloading JSON metadata
[CeskaTelevize] iFramePlayer.php: Downloading JSON metadata
[CeskaTelevize] iFramePlayer.php: Downloading JSON metadata
[CeskaTelevize] iFramePlayer.php: Downloading JSON metadata
ERROR: No video formats found; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "/usr/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 815, in wrapper
    return func(self, *args, **kwargs)
  File "/usr/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 836, in __extract_info
    ie_result = ie.extract(url)
  File "/usr/bin/youtube-dl/youtube_dl/extractor/common.py", line 534, in extract
    ie_result = self._real_extract(url)
  File "/usr/bin/youtube-dl/youtube_dl/extractor/ceskatelevize.py", line 201, in _real_extract
    self._sort_formats(e['formats'])
  File "/usr/bin/youtube-dl/youtube_dl/extractor/common.py", line 1374, in _sort_formats
    raise ExtractorError('No video formats found')
youtube_dl.utils.ExtractorError: No video formats found; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.

~@kusansky$
```

## Description
Yes, this video works in Firefox on my desktop, but it just doesn't download at all with youtube-dl.

**Comment by dirkf** [comment]
Pořad bylo možné přehrát do 16. 7. 2021 but probably would have been closed by the linked PR.

**Comment by mcepl** [comment]
> "Pořad bylo možné přehrát do 16. 7. 2021"

That’s “Show has been possible to play until 16. 7. 2021.”

**Comment by dirkf** [comment]
Just as I inferred by analogy with another **special** slavonic language. We can't tell if the show would have been accessible with the PR version of the extractor.

**Comment by mcepl** [comment]
This one should be more stable (originally from 2015 and it is still playable):
```
stitny~$ youtube-dl --verbose 'https://www.ceskatelevize.cz/porady/11003016190-karel-havlicek-borovsky/'
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['--verbose', 'https://www.ceskatelevize.cz/porady/11003016190-karel-havlicek-borovsky/']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, pref UTF-8
[debug] youtube-dl version 2021.12.17
[debug] Python version 3.8.12 (CPython) - Linux-5.16.10-1-default-x86_64-with-glibc2.34
[debug] exe versions: ffmpeg 4.4.1, ffprobe 4.4.1
[debug] Proxy map: {}
[CeskaTelevizePorady] 11003016190-karel-havlicek-borovsky: Downloading webpage
ERROR: Unable to extract iframe player url; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "/usr/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 815, in wrapper
    return func(self, *args, **kwargs)
  File "/usr/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 836, in __extract_info
    ie_result = ie.extract(url)
  File "/usr/bin/youtube-dl/youtube_dl/extractor/common.py", line 534, in extract
    ie_result = self._real_extract(url)
  File "/usr/bin/youtube-dl/youtube_dl/extractor/ceskatelevize.py", line 282, in _real_extract
    data_url = update_url_query(unescapeHTML(self._search_regex(
  File "/usr/bin/youtube-dl/youtube_dl/extractor/common.py", line 1012, in _search_regex
    raise RegexNotFoundError('Unable to extract %s' % _name)
youtube_dl.utils.RegexNotFoundError: Unable to extract iframe player url; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.

stitny~$
```

**Comment by dirkf** [comment]
That's OK with new code:
```
$ git checkout dlp-ceskatv-extractor-backport
Switched to branch 'dlp-ceskatv-extractor-backport'
$ python -m youtube_dl -F -v 'https://www.ceskatelevize.cz/porady/11003016190-karel-havlicek-borovsky/'
[debug] System config: [u'--prefer-ffmpeg']
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: [u'-F', u'-v', u'https://www.ceskatelevize.cz/porady/11003016190-karel-havlicek-borovsky/']
[debug] Encodings: locale UTF-8, fs UTF-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2021.12.17
[debug] Git HEAD: d799c4050
[debug] Python version 2.7.17 (CPython) - Linux-4.4.0-210-generic-i686-with-Ubuntu-16.04-xenial
[debug] exe versions: avconv 4.3, avprobe 4.3, ffmpeg 4.3, ffprobe 4.3
[debug] Proxy map: {}
[CeskaTelevize] 11003016190-karel-havlicek-borovsky: Downloading webpage
[CeskaTelevize] 11003016190-karel-havlicek-borovsky: Getting IFRAME hash
[CeskaTelevize] 11003016190-karel-havlicek-borovsky: Downloading player
[CeskaTelevize] 11003016190-karel-havlicek-borovsky: Downloading JSON metadata
[CeskaTelevize] 11003016190-karel-havlicek-borovsky: Downloading JSON metadata
[CeskaTelevize] 11003016190-karel-havlicek-borovsky: Downloading MPD manifest
[CeskaTelevize] 11003016190-karel-havlicek-borovsky: Downloading MPD manifest
[CeskaTelevize] 11003016190-karel-havlicek-borovsky: Downloading JSON metadata
[CeskaTelevize] 11003016190-karel-havlicek-borovsky: Downloading JSON metadata
[CeskaTelevize] 11003016190-karel-havlicek-borovsky: Downloading m3u8 information
[CeskaTelevize] 11003016190-karel-havlicek-borovsky: Downloading m3u8 information
[info] Available formats for 61924494877211424:
format code                      extension  resolution note
hls-audioDescription-128         mp4        audio only  128k , mp4a.40
dash-audioDescription-1002-1507  m4a        audio only [que] DASH audio  128k , m4a_dash container, mp4a.40.2 (48000Hz)
dash-main-1002-1504              m4a        audio only [ces] DASH audio  128k , m4a_dash container, mp4a.40.2 (48000Hz)
dash-main-1001-1502              mp4        512x288    DASH video  500k , mp4_dash container, avc1.4d4015, video only
dash-audioDescription-1001-1507  mp4        720x404    DASH video 1024k , mp4_dash container, avc1.4d401e, video only
dash-main-1001-1503              mp4        720x404    DASH video 1032k , mp4_dash container, avc1.4d401e, video only
dash-main-1001-1504              mp4        1024x576   DASH video 2048k , mp4_dash container, avc1.4d401f, video only
dash-main-1001-1505              mp4        1280x720   DASH video 3584k , mp4_dash container, avc1.4d401f, video only
hls-main-628                     mp4        unknown     628k 
hls-audioDescription-1152        mp4        unknown    1152k 
hls-main-1160                    mp4        unknown    1160k 
hls-main-2176                    mp4        unknown    2176k 
hls-main-3712                    mp4        unknown    3712k  (best)
$
```

**Comment by mcepl** [comment]
I see, so I am just waiting on new release.
